dagger-core/src/main/java/io/odpf/dagger/core/source/parquet/reader/ParquetReader.java

```diff
@@ -0,0 +1,94 @@
+package io.odpf.dagger.core.source.parquet.reader;
+
+import io.odpf.dagger.common.serde.parquet.deserialization.SimpleGroupDeserializer;
+import io.odpf.dagger.core.exception.ParquetFileSourceReaderInitializationException;
+import org.apache.flink.connector.file.src.reader.FileRecordFormat;
+import org.apache.flink.types.Row;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.column.page.PageReadStore;
+import org.apache.parquet.example.data.Group;
+import org.apache.parquet.example.data.simple.SimpleGroup;
+import org.apache.parquet.example.data.simple.convert.GroupRecordConverter;
+import org.apache.parquet.hadoop.ParquetFileReader;
+import org.apache.parquet.hadoop.util.HadoopInputFile;
+import org.apache.parquet.io.ColumnIOFactory;
+import org.apache.parquet.io.MessageColumnIO;
+import org.apache.parquet.io.RecordReader;
+import org.apache.parquet.schema.MessageType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+
+public class ParquetReader implements FileRecordFormat.Reader<Row> {
+    private final Path hadoopFilePath;
+    private final SimpleGroupDeserializer simpleGroupDeserializer;
+    private long currentRecordIndex;
+    private final ParquetFileReader parquetFileReader;
+    private long rowCount;
+    private RecordReader<Group> recordReader;
+    private final MessageType schema;
+    private static final Logger LOGGER = LoggerFactory.getLogger(ParquetReader.class.getName());
+
+    private ParquetReader(Path hadoopFilePath, SimpleGroupDeserializer simpleGroupDeserializer, ParquetFileReader parquetFileReader) throws IOException {
+        this.hadoopFilePath = hadoopFilePath;
+        this.simpleGroupDeserializer = simpleGroupDeserializer;
+        this.parquetFileReader = parquetFileReader;
+        this.schema = this.parquetFileReader.getFileMetaData().getSchema();
+        changeReaderPosition(this.parquetFileReader.readNextRowGroup());
```

Comment:
We are initialising the reader position here so that when Flink calls `read` internally in the next step, it reads from the beginning of the file. 
I don't think this will be computationally expensive. Additionally, this is initialization code which should be in the constructor I think.

Reply:
It will be faster than the network. However, still it is reading all the chunks of the file which will be slower. Would prefer it be initialized lazily.

Reply:
@gauravsinghania Refactored as part of commit https://github.com/odpf/dagger/pull/121/commits/0c25068fb7a44fdf1d46d548ddbc6ef81e3f7701